### PR TITLE
Indent group questions in form summary and app diff views

### PR DESCRIPTION
**Pull request by proteusvacuum**
https://dimagi-dev.atlassian.net/browse/PF-313

Screenshot from 2019-04-30 17-34-57

Screenshot from 2019-04-30 17-36-32

There's currently an issue that when you have a group that only has "hidden" items (save to case / hidden values) that don't indent properly because of the way our xform parser figures out what groups exist in a form. (https://github.com/dimagi/commcare-hq/blob/fr%2Findent-group-questions/corehq/apps/app_manager/xform.py#L1062-L1067) 
I'm not really sure how to solve that right now, and that function is very gnarly, Very Important, and very lightly tested.

### CODE REVIEW

File: corehq/apps/app_manager/xform.py [hunk 1]
```diff
@@ -1131,7 +1132,7 @@ def _get_select_question_option(item):
                         "tag": "hidden",
                         "value": '{}/@{}'.format(path, attrib),
                         "repeat": node_info['repeat'],
-                        "group": node_info['repeat'],
+                        "group": node_info['group'],
```
Comment (orangejenny) [inline corehq/apps/app_manager/xform.py]: Was this causing a problem previously or just being ignored?

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/app_manager/templates/app_manager/form_summary_base.html [hunk 2]
```diff
@@ -4,172 +4,172 @@
 
 <!-- ko if: errors.length -->
 <div class="alert alert-danger">
-    <i class="fa fa-exclamation-circle"></i>
```
[the remark this answers is not in the document]
Comment (proteusvacuum) [reply corehq/apps/app_manager/templates/app_manager/form_summary_base.html]: yes: 
Screenshot from 2019-05-01 13-04-23

File: corehq/apps/app_manager/templates/app_manager/form_summary_base.html [hunk 3]
```diff
@@ -173,3 +92,93 @@ <h5>
     &nbsp;<span data-bind="text: text"></span>
   </button>
 </script>
+
+<script type="text/html" id="form-question">
+  <li data-bind="visible: isVisible, css: getDiffClass('question') ">
+    <i data-bind="attr: { 'class': $root.questionIcon($data), 'title': type }"></i>
+    <!-- ko ifnot: $root.readOnly -->
```
[the remark this answers is not in the document]
Comment (proteusvacuum) [reply corehq/apps/app_manager/templates/app_manager/form_summary_base.html]: I do 100% agree. Will do that in a followup.

File: corehq/apps/app_manager/xform.py (hunk 1 above)
Comment (proteusvacuum) [reply corehq/apps/app_manager/xform.py]: I don't think anything was using this information previously...

**Comment by orangejenny** [comment]
Cool, thanks for explanations.

I went to kick the build, but test failures look real.

**Comment by proteusvacuum** [comment]
I think d357297 fixes the known issue I mentioned in the original comment.

**Review by orangejenny — APPROVED** [review]

**Pull request merged**